src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/AndroidCertificatePal.cs

```diff
@@ -0,0 +1,441 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Formats.Asn1;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Asn1;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed class AndroidCertificatePal : ICertificatePal
+    {
+        private SafeX509Handle _cert;
+        private SafeEvpPKeyHandle? _privateKey;
+
+        private CertificateData _certData;
+
+        public static ICertificatePal FromHandle(IntPtr handle)
+        {
+            if (handle == IntPtr.Zero)
+                throw new ArgumentException(SR.Arg_InvalidHandle, nameof(handle));
+
+            var newHandle = new SafeX509Handle(Interop.JObjectLifetime.NewGlobalReference(handle));
+            return new AndroidCertificatePal(newHandle);
+        }
+
+        public static ICertificatePal FromOtherCert(X509Certificate cert)
+        {
+            Debug.Assert(cert.Pal != null);
+
+            // Ensure private key is copied
+            AndroidCertificatePal certPal = (AndroidCertificatePal)cert.Pal;
+            return certPal.DuplicateHandles();
+        }
+
+        public static ICertificatePal FromBlob(ReadOnlySpan<byte> rawData, SafePasswordHandle password, X509KeyStorageFlags keyStorageFlags)
+        {
+            // TODO: [AndroidCrypto] Handle PKCS#12
```

Comment:
The good news is that macOS and Unix-OpenSSL both use a managed PKCS#12 reader/writer to take away a lot of corner case inconsistencies.

But it probably requires that there already be support from the Algorithms layer :smile: